src/servant/tt_001/F84.c

```diff
@@ -520,7 +520,91 @@ void func_us_801720D4(void) {}
 
 void func_us_801720DC(void) {}
 
-INCLUDE_ASM("servant/tt_001/nonmatchings/F84", func_us_801720E4);
+extern Primitive* D_us_801737F0;
+extern s32 D_us_801737E0;
+extern s32 D_us_801737E4;
+extern s32 D_us_801737F4;
+extern s32 D_us_801737F8;
+extern s32 D_us_801737E8;
+extern s32 D_us_801737EC;
+
+void func_us_801720E4(Entity* self) {
+
+    if (self->params != 0) {
+        DestroyEntity(self);
+        return;
+    }
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, 8);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->flags =
+            FLAG_KEEP_ALIVE_OFFCAMERA | FLAG_HAS_PRIMS | FLAG_UNK_20000;
+
+        D_us_801737F0 = &g_PrimBuf[self->primIndex];
+
+        for (D_us_801737E0 = 0; D_us_801737E0 < 8; D_us_801737E0++) {
+            D_us_801737F0->tpage = 0x1A;
+            D_us_801737F0->clut = 0x143;
+
+            D_us_801737F0->u0 = D_us_801737F0->u2 = 0x78;
+            D_us_801737F0->u1 = D_us_801737F0->u3 = 0x80;
+
+            D_us_801737F0->v0 = D_us_801737F0->v1 = 0x30;
+            D_us_801737F0->v2 = D_us_801737F0->v3 = 0x38;
+
+            D_us_801737F0->r0 = D_us_801737F0->r1 = D_us_801737F0->r2 =
+                D_us_801737F0->r3 = D_us_801737F0->g0 = D_us_801737F0->g1 =
+                    D_us_801737F0->g2 = D_us_801737F0->g3 = D_us_801737F0->b0 =
+                        D_us_801737F0->b1 = D_us_801737F0->b2 =
+                            D_us_801737F0->b3 = 0x80;
+
+            D_us_801737F0->priority = self->zPriority + 1;
+            D_us_801737F0->blendMode = 0x135;
+
+            D_us_801737F0 = D_us_801737F0->next;
+        }
+
+        D_us_801737F4 = 0x1E;
+        D_us_801737F8 = 0;
+        g_api.PlaySfx(SFX_GLASS_SHARDS);
+        self->step++;
+
+    case 1:
+        D_us_801737F8 = (D_us_801737F8 + 0x32) & 0xFFF;
+        D_us_801737F4--;
+        if (D_us_801737F4 < 0) {
+            DestroyEntity(self);
+            return;
+        }
+    }
+    self->posX.i.hi = self->ext.factory.parent->posX.i.hi;
+    self->posY.i.hi = self->ext.factory.parent->posY.i.hi;
```

Comment:
Oh also, we shouldn't be using `ext.factory` for entities that aren't factories.

Reply:
How to properly write it then?

Reply:
Add a new entry to the Ext union in `include/entity.h` and use that here, until we can learn more about what this function you've decompiled actually is.

Reply:
Thanks. How to properly name it then?

Reply:
You can just name it after the name of this function, look at the other entries in Ext, there are several with numerical names for the unknown functions they're used in

Reply:
I can't get an OK with this changes

Reply:
I may already have an ext for this.  Let me check.  I just don't know which Entity* is the correct one.  The ext is self->ext.ghost

Reply:
Unfortunately that won't work. factory.parent is at offset 0x8C, but that offset is already used in ext.ghost. The entity here is of course related to the ghost, but is not the ghost itself so it uses a different Ext.

Reply:
Dang.  Yeah, ET_Ghost uses 0x8C as a cycle counter for the shimmer effect (I think).
Looks like there needs to be an entity like batFamBlueTrail, but generic for servants, or the events that servant entities generate such as in [CreateEventEntity](https://github.com/Xeeynamo/sotn-decomp/blob/11631e1475691c68acacedd452c666a7364bd1ea/src/servant/create_event_entity.h#L22)

Reply:
For now I think it's best to have different Ext entries for different entities. Once we have all the entities decompiled, we can figure out what can be consolidated, and what patterns exist to indicate that two entities are the same.

Reply:
In the mean time, @mrmidi if you make a copy of [ET_BatFamBlueTrail](https://github.com/Xeeynamo/sotn-decomp/blob/11631e1475691c68acacedd452c666a7364bd1ea/include/entity.h#L661) and call it something like ET_GhostEvent or something like that, add it to the [Ext Union](https://github.com/Xeeynamo/sotn-decomp/blob/11631e1475691c68acacedd452c666a7364bd1ea/include/entity.h#L1893C16-L1893C17), you should be able to replace factory with ghostEvent (or whatever you name it in the union.